cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2757 @@
+    case Type::LocalDateTime:
+      return mgp::type_local_date_time();
+    case Type::Duration:
+      return mgp::type_duration();
+    default:
+      break;
+  }
+  throw ValueException("Unknown type error!");
+}
+
+/// @brief Converts C++ API list types to their MGP API equivalents.
+/// @note In the MGP API, it is necessary to define the list items’ type.
+inline mgp_type *ToMGPType(Type list_type, Type item_type) {
+  if (list_type != Type::List) {
+    throw ValueException("Unknown type error!");
+  }
+
+  switch (item_type) {
+    case Type::Bool:
+      return mgp::type_list(mgp::type_bool());
+    case Type::Int:
+      return mgp::type_list(mgp::type_int());
+    case Type::Double:
+      return mgp::type_list(mgp::type_float());
+    case Type::String:
+      return mgp::type_list(mgp::type_string());
+    case Type::Map:
+      return mgp::type_list(mgp::type_map());
+    case Type::Node:
+      return mgp::type_list(mgp::type_node());
+    case Type::Relationship:
+      return mgp::type_list(mgp::type_relationship());
+    case Type::Path:
+      return mgp::type_list(mgp::type_path());
+    case Type::Date:
+      return mgp::type_list(mgp::type_date());
+    case Type::LocalTime:
+      return mgp::type_list(mgp::type_local_time());
+    case Type::LocalDateTime:
+      return mgp::type_list(mgp::type_local_date_time());
+    case Type::Duration:
+      return mgp::type_list(mgp::type_duration());
+    default:
+      break;
+  }
+  throw ValueException("Unknown type error!");
+}
+
+/// @brief Converts MGP API types to their C++ API equivalents.
+inline Type ToAPIType(mgp_value_type type) {
+  switch (type) {
+    case MGP_VALUE_TYPE_NULL:
+      return Type::Null;
+    case MGP_VALUE_TYPE_BOOL:
+      return Type::Bool;
+    case MGP_VALUE_TYPE_INT:
+      return Type::Int;
+    case MGP_VALUE_TYPE_DOUBLE:
+      return Type::Double;
+    case MGP_VALUE_TYPE_STRING:
+      return Type::String;
+    case MGP_VALUE_TYPE_LIST:
+      return Type::List;
+    case MGP_VALUE_TYPE_MAP:
+      return Type::Map;
+    case MGP_VALUE_TYPE_VERTEX:
+      return Type::Node;
+    case MGP_VALUE_TYPE_EDGE:
+      return Type::Relationship;
+    case MGP_VALUE_TYPE_PATH:
+      return Type::Path;
+    case MGP_VALUE_TYPE_DATE:
+      return Type::Date;
+    case MGP_VALUE_TYPE_LOCAL_TIME:
+      return Type::LocalTime;
+    case MGP_VALUE_TYPE_LOCAL_DATE_TIME:
+      return Type::LocalDateTime;
+    case MGP_VALUE_TYPE_DURATION:
+      return Type::Duration;
+    default:
+      break;
+  }
+  throw ValueException("Unknown type error!");
+}
+}  // namespace util
+
+/* #region Graph (Id, Graph, Nodes, GraphRelationships, Relationships, Properties & Labels) */
+
+// Graph:
+
+int64_t Graph::order() const {
+  int64_t i = 0;
+  for (const auto &v : nodes()) {
+    i++;
+  }
+  return i;
+}
+
+int64_t Graph::size() const {
+  int64_t i = 0;
+  for (const auto &_ : relationships()) {
+    i++;
+  }
+  return i;
```

Comment:
Can we use `std::distance` https://en.cppreference.com/w/cpp/iterator/distance?

Reply:
`std::distance` keeps incrementing the iterator until it reaches the `.end()`; with some iterators as this one this causes an error because incrementing them doesn’t eventually reach it (`.end()` is defined as returning `nullptr`)